CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guide
+
+Welcome to the project, and thanks for considering contributing to this project. 
+This document should answer many questions you may have about contributing to this project.
+
+If you have any questions or need clarifications on topics covered here, please feel free to reach out to us on the [#cnoe-interest](https://cloud-native.slack.com/archives/C05TN9WFN5S) channel on CNCF Slack.
+
+## Setting up a development environment
+
+To get started with the project on your machine, you need to install the following tools:
+1. Go 1.21+. See [this official guide](https://go.dev/doc/install) from Go authors.
+2. Make. You can install it through a package manager on your system. E.g. Install `build-essential` for Ubuntu systems.
+3. Docker. Similar to Make, you can install it through your package manager.
```

Comment:
I didn't want to mention podman because we don't support it yet :(. I personally would love to support podman. Dcoker desktop kills my laptop battery.